projects/media-viewer/src/lib/utils/mouse-selection-utils.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MouseSelectionUtils {
+
+  // Return the absolute positioning of the top, left x & y coordinates relative to the localElement
+  static getTargetElementAbsoluteMouseCoordinates(event: MouseEvent): { x: number, y: number } {
+    const responseObject = {
+      x: event.x,
+      y: event.y
+    };
+
+    // Get the VIEWPORT-relative coordinates of the click.
+    const viewportX: number = event.clientX;
+    const viewportY: number = event.clientY;
+    //
+    // Obtain the HTMLElement from the event
+    const localElement: Element = (<Element>event.target) || (<Element>event.srcElement);
```

Comment:
I'm curious about this - given you have the type defined on the declaration I would think TypeScript can infer it. Does it work as:

`const localElement: Element = event.target || event.srcElement;`

Reply:
Ha!
Well, i've had to cast it back to an Element interface so that I can call the subsequent methods. The native returned Element is actually an HTML Element. 

I probably don't need the : Element aspect though

Reply:
Okay, just a heads up - type casting in TypeScript doesn't effect runtime like Java's does. It just gives the compiler more information

Reply:
okay thanks, yeah i can see the js that's generated beneath. Just wanted to please the Typescript compiler ;) will review and look again if i can work it without it